src/lib/chat/index.ts

```diff
@@ -54,6 +54,7 @@ export interface IChatClient {
     optimisticId?: string
   ) => Promise<MessagesResponse>;
   fetchConversationsWithUsers: (users: User[]) => Promise<Partial<Channel>[]>;
+  deleteMessageByRoomId: (channelId: string, messageId: number) => Promise<any>;
```

Comment:
1. The argument should probably be `roomId` :)
2. `messageId` should be a string. They aren't numbers.
3. Return type should probably be `Promise<void>` if the return value is never used.

Reply:
Nice will update.

Changing `messageId` to `string` will mean I need to update the following `messageId` to a string in the sendbird client.

```
   onMessageDeleted: (channel, messageId) => {
        const channelId = this.getChannelId(channel);

        events.receiveDeleteMessage(channelId, messageId.toString());
      },
```

Reply:
Actually I'm confused here. Isn't the sendbird messageId type `number` ?

The api looked like this:
```
export async function deleteMessageApi(channelId: string, messageId: number): Promise<number> {
  const response = await del<any>(`/chatChannels/${channelId}/message`).send({ message: { id: messageId } });

  return response.status;
}
```

then in the `deleteMessage` saga we're changing to string:
```
const messageIdsToDelete = fullMessages.filter((m) => m.rootMessageId === messageId.toString()).map((m) => m.id); // toString() because message ids are currently a number
```

Reply:
> Isn't the sendbird messageId type number ?

It is usually a number but we never should have defined it as such. It should always have been a string. Someone messed up the original model so we're stuck with juggling it. Since we're moving towards Matrix we should keep the matrix client correct and fix everything else.

Reply:
Sounds good thanks! Ive changed all the types to string